cmd/deploy.go

```diff
@@ -139,43 +139,63 @@ func deploy(environmentType string, serviceName string, serviceType string, fold
 	}
 	s1.Stop()
 
-	s2 := spinner.New(spinner.CharSets[26], 100*time.Millisecond)
-
 	//staging the deploy, total time is about 2100
-	progressBar("Initializing deployment... ", 700)
-	fmt.Print("\n")
-	progressBar("Deploying service ...      ", 700)
-	fmt.Print("\n")
-	progressBar("Finalizing deployment...   ", 700)
-	fmt.Print("\n")
+	// progressBar("Initializing deployment... ", 700)
+	// fmt.Print("\n")
+	// progressBar("Deploying service ...      ", 700)
+	// fmt.Print("\n")
+	// progressBar("Finalizing deployment...   ", 700)
+	// fmt.Print("\n")
 
 	servUrl := ""
-	s2.Prefix = "Fetching service URL... "
+
+	p := mpb.New(mpb.WithWidth(64))
+	defer p.Wait()
+	total := 68
+
+	bar := getProgressBar(p, "Deploying service ...", total)
 
 	for {
 		update, err := stream.Recv()
 		if err == io.EOF {
 			break
 		} else if err != nil {
+			// bar.Abort(true)
+			//s2.Stop()
 			log.Fatalf("server side error: %s", err.Error())
 		}
 
-		if msg := update.GetDeploymentUpdate(); msg != nil {
+		// bar.Increment()
+		deployUpdate := update.GetDeploymentUpdate()
+		if deployUpdate != nil {
+			bar.IncrBy(2) // there are 34 msgs the server send to the client, so total is 68/34 = 2 per message will get us to 100
 			continue
 		}
 
 		servUrl = update.GetURL()
-
-		if servUrl == "" {
-			s2.Start()
-			time.Sleep(10 * time.Second)
-		} else {
-			s2.Stop()
-			fmt.Printf("Service is deployed at: %s\n", servUrl)
+		barStatus := bar.Completed()
+
+		if servUrl != "" {
+			if !barStatus {
+				rem := 68 - bar.Current()
+				fmt.Println("current bar is at ", bar.Current())
+				fmt.Println("remaining is", rem) //if servUrl is returned but bar isn't done yet, just increment it so that it looks smooth
+				for i := 0; i <= int(rem); i++ {
+					time.Sleep(time.Duration(rand.Intn(10)+1) * time.Duration(rem) / 10)
+					bar.IncrBy(2)
+				}
+				fmt.Printf("Service is deployed at: %s\n", servUrl)
+				break
+			}
 		}
-		break
-	}
 
+		fmt.Printf("Service is deployed at: %s\n", servUrl)
```

Comment:
There is an error case that happens on the server but it still sends back data. I had a small WIP branch where the "GetDeploymentUpdate" actually sent back if it was a failure message or not. But didnt continue with it.

Right now this code is kinda weird because we shouldnt be printing this line if there is no server url, but instead the error of what went wrong. Or at least that _something went wrong_ and it should be reported or we should receive an alert of it or something..